insonmnia/worker/server.go

```diff
@@ -607,7 +610,7 @@ func (m *Worker) taskAllowed(ctx context.Context, request *pb.StartTaskRequest)
 	if err != nil {
 		return false, nil, err
 	}
-	if level <= pb.IdentityLevel_REGISTERED {
+	if level <= m.cfg.Whitelist.PrivilegedIdentityLevel {
```

Comment:
Seems that only users which level is `less` than required should be checked with the allowed list.